### PR TITLE
Develop 

### PR DESCRIPTION
**Pull request by darkbizarre**
Add:
- external dependencies
- socket server
- socket login for angular app
- appointment page with logic for Appointmet Call

### CODE REVIEW

**Comment by mcajhill** [comment]
I've noticed a couple of things looking at the code:

1) Doing an npm start socket-server doesn't work because there isn't a start script in the root directory of the project, and no reference to a script called "start" in package.json

2) You are using ES6, which is fine, but the application is written in JS 1.5. So I would have expected to see a transpiler to compile the JS down to 1.5. However, the UI project is currently undergoing a refactoring process which will see the entire UI written in ES6 so this isn't a huge issue so long as deployment isn't hindered.

3) I don't see any documentation on the deployment requirements now - certain questions should be answered, e.g. Have the deployment requirements changed? Does the server need NodeJS installed? A certain version of NodeJS? Do the Tomcat startup scripts need to change? etc. etc.

4) You are opening up database connections via Node, whereas we have a middleware solution which is designed to handle data access. Keeping the connection details in there for now isn't a huge issue, and those connections can be refactored. However, if they were would that affect the functionality of your solution?

**Comment by mcajhill** [comment]
A couple of other things too:

5) Does the "node .\socket-server\socket.js" isn't working either. Does it rely on a certain version of NodeJS? I get "SyntaxError: Use of const in strict mode."

6) You have the appointment.html file in the root directory of the app. Is there a reason for it being here instead of in the views folder? It also doesn't follow the view injection style of the Angular app. Is there a reason for this? Does it have to be a fully formed HTML document?

**Comment by aladdi1n** [comment]
1) Doing an npm start socket-server doesn't work because there isn't a start script in the root directory of the project, and no reference to a script called "start" in package.json
- fixed

2) You are using ES6, which is fine, but the application is written in JS 1.5. So I would have expected to see a transpiler to compile the JS down to 1.5. However, the UI project is currently undergoing a refactoring process which will see the entire UI written in ES6 so this isn't a huge issue so long as deployment isn't hindered.
- this wasn't mentioned in requirements. Is it really necessary to transpile server into js 1.5? It doesn't influence front end 

3) I don't see any documentation on the deployment requirements now - certain questions should be answered, e.g. Have the deployment requirements changed? Does the server need NodeJS installed? A certain version of NodeJS? Do the Tomcat startup scripts need to change? etc. etc.
- Server requires Node.js version >= 6.0.0. Other dependencies are into package.json and they will be set after npm install

4) You are opening up database connections via Node, whereas we have a middleware solution which is designed to handle data access. Keeping the connection details in there for now isn't a huge issue, and those connections can be refactored. However, if they were would that affect the functionality of your solution?
- We've requested earlier a few additional API methods, but we've got the feedback from Tony, that it is too hard and too long to get them, so there was a decision to make it any possible way just on our side. If you decide to change that, yes, a few small changes would be needed on our side too.

5) Does the "node .\socket-server\socket.js" isn't working either. Does it rely on a certain version of NodeJS? I get "SyntaxError: Use of const in strict mode."
- Requires Node.js version >= 6.0.0

6) You have the appointment.html file in the root directory of the app. Is there a reason for it being here instead of in the views folder? It also doesn't follow the view injection style of the Angular app. Is there a reason for this? Does it have to be a fully formed HTML document?
- Yes, it is a separate page like index.html or 404.html, not just a view for angular, it didn't make sense to use it for a simple page

**Readme file is updated https://yadi.sk/i/vemjm7OFuUYbh**

File: app/views/appointments/appointments-detail.html [hunk 1]
```diff
@@ -64,9 +64,22 @@
...
+            <li ng-repeat="message in messages">{{message.timestamp}} - {{message.author}}{{message.message}}</li>
+          </ul>
+        </div>
+      </div>
     </div>
   </div>
 </div>
 <div class="form-group form-group-sm text-right" ng-if="currentUser.permissions.indexOf('WRITE') !== -1">
+  <button class="btn btn-primary" ng-click="startAppointment();" ng-if="!isClosed && !$root.socketData.showJoinAppointment">Start Appointment</button>
+  <button class="btn btn-primary" ng-click="joinAppointment();" ng-if="!isClosed && $root.socketData.showJoinAppointment == appointment.sourceId">Join Appointment</button>
```
Comment (mcajhill) [inline app/views/appointments/appointments-detail.html]: Could you add an ng-cloak to this element to stop it flickering as the page loads before the CSS class is applied (if the CSS then hides it).

File: app/views/appointments/appointments-detail.html [hunk 2]
```diff
@@ -64,9 +64,22 @@
...
+          <ul class="list-unstyled list-messages">
+            <li ng-repeat="message in messages">{{message.timestamp}} - {{message.author}}{{message.message}}</li>
+          </ul>
+        </div>
+      </div>
     </div>
   </div>
 </div>
 <div class="form-group form-group-sm text-right" ng-if="currentUser.permissions.indexOf('WRITE') !== -1">
+  <button class="btn btn-primary" ng-click="startAppointment();" ng-if="!isClosed && !$root.socketData.showJoinAppointment">Start Appointment</button>
```
Comment (mcajhill) [inline app/views/appointments/appointments-detail.html]: Could you add an ng-cloak to this element to stop it flickering as the page loads before the CSS class is applied (if the CSS then hides it).

**Pull request merged**